shared/src/background.js

```diff
@@ -0,0 +1,185 @@
+import { summarizeContent } from './lib/utils.js';
+
+if (!globalThis.browser) {
+  globalThis.browser = chrome;
+}
+
+let sessionToken = undefined;
+let syncSessionFromExisting = true;
+let sessionApiToken = undefined;
+let sessionApiEngine = undefined;
+let IS_CHROME = true;
+
+// Very hacky, but currently works flawlessly
+if (typeof browser.runtime.getBrowserInfo === 'function') {
+  IS_CHROME = false;
+}
```

Comment:
This is currently the easiest/simplest way to determine if the browser is `firefox` or not (more like is it not `chromium`), since [only `firefox` supports it](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/Runtime/getBrowserInfo#browser_compatibility).